components/concept/engine/src/main/java/mozilla/components/concept/engine/Settings.kt

```diff
@@ -40,6 +41,11 @@ abstract class Settings {
      */
     open var requestInterceptor: RequestInterceptor? by UnsupportedSetting()
 
+    /**
+     * Setting to provide a history delegate to the engine.
+     */
+    open var historyTrackingDelegate: HistoryTrackingDelegate? by UnsupportedSetting()
```

Comment:
I assume you went with a delegate in `Settings` instead of exposing that in the `EngineSession.Observer` because you want a single observer for all engine sessions?

Reply:
Basically, yes. This seemed like the most straightforward approach once you start wiring things together, configuring a delagate this way becomes trivial.

Going through enginesession would also basically mean going through the SessionManager, I think, and that didn’t seem like the right fit here.